File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/interfaces/LatexInterface.java
package xyz.jienan.xkcd.whatif.interfaces;

import android.webkit.JavascriptInterface;

public class LatexInterface {

    private boolean canParentScroll = true;

    @JavascriptInterface
    public void onTouch(int i) {
        canParentScroll = i == 3;
    }

    public boolean canScrollHor() {
        return canParentScroll;
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/interfaces/LatexInterface.kt
package xyz.jienan.xkcd.whatif.interfaces

import android.webkit.JavascriptInterface

class LatexInterface {

    private var canParentScroll = true

    @JavascriptInterface
    fun onTouch(i: Int) {
        canParentScroll = i == 3
    }

    fun canScrollHor() = canParentScroll
}
